packages/internal-testing-tools/android/src/test/kotlin/com/datadog/reactnative/internaltesting/DdSessionReplayImplementationTest.kt

```diff
@@ -0,0 +1,119 @@
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative.internaltesting
+
+import android.content.Context
+import androidx.annotation.WorkerThread
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.context.DatadogContext
+import com.datadog.android.api.feature.Feature
+import com.datadog.android.api.feature.FeatureScope
+import com.datadog.android.api.storage.DataWriter
+import com.datadog.android.api.storage.EventBatchWriter
+import com.datadog.android.api.storage.RawBatchEvent
+import com.datadog.android.core.InternalSdkCore
+import com.datadog.android.core.persistence.Serializer
+import com.datadog.android.core.persistence.serializeToByteArray
+import com.datadog.reactnative.DatadogSDKWrapperStorage
+import com.facebook.react.bridge.Promise
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+internal class DdInternalTestingImplementationTest {
+
+    lateinit var testedInternalTesting: DdInternalTestingImplementation
+
+    @Mock
+    lateinit var mockPromise: Promise
+
+    @Mock
+    lateinit var mockCore: InternalSdkCore
+
+    @Mock
+    lateinit var mockContext: DatadogContext
+
+    @BeforeEach
+    fun `set up`() {
+        testedInternalTesting =
+            DdInternalTestingImplementation()
+    }
+
+    @AfterEach
+    fun `tear down`() {
+    }
+
+    @Test
+    fun `M return captured events W enable()`() {
+        // Given
+        testedInternalTesting.enable(mockPromise)
+        val mockFeature = MockFeature("mockFeature")
+        val mockFeatureScope = MockFeatureScope(mockFeature)
+        whenever(mockCore.getFeature(mockFeature.name)).doReturn(
+            mockFeatureScope
+        )
+        whenever(mockCore.getDatadogContext()).doReturn(
+            mockContext
+        )
+        DatadogSDKWrapperStorage.setSdkCore(mockCore)
+        DatadogSDKWrapperStorage.notifyOnInitializedListeners()
+
+
+        // When
+        val wrappedCore = DatadogSDKWrapperStorage.getSdkCore() as StubSDKCore
+        wrappedCore.registerFeature(mockFeature)
+        wrappedCore.getFeature(mockFeature.name)!!.withWriteContext { _, eventBatchWriter ->
+            MockDataWriter().write(eventBatchWriter, MockEvent("mock event for test"))
```

Comment:
it seems you need need `MockDataWriter` and `MockEvent` here, it has no purpose. This line can be written directly as:

```suggestion
            eventBatchWriter.write(RawBatchEvent(data = "mock event for test".toByteArray()), batchMetadata = null)
```